benchmark/src/benchmark-step.ts

```diff
@@ -0,0 +1,48 @@
+import type { BenchmarkValue } from './benchmark';
+import { benchmark } from './benchmark';
+import { print } from './print';
+import { Step } from './step';
+
+/**
+ * A benchmark step is time measured and run multiple times
+ * The printed out value is the time it took to run the benchmark
+ */
+export class BenchmarkStep<
+    State extends { [key in Name]?: BenchmarkValue<any> | undefined },
+    Name extends string & keyof State = string & keyof State,
+    Value extends NonNullable<State[Name]> extends BenchmarkValue<infer T>
+        ? T
+        : never = NonNullable<State[Name]> extends BenchmarkValue<infer T>
+        ? T
+        : never
+> extends Step<State, Name, State[Name]> {
```

Comment:
```suggestion
    State extends { [key in Name]?: BenchmarkValue<Value> | undefined },
    Name extends string & keyof State = string & keyof State,
    Value extends NonNullable<State[Name]> extends BenchmarkValue<infer T>
        ? T
        : never = NonNullable<State[Name]> extends BenchmarkValue<infer T>
        ? T
        : never
> extends Step<{ [Key in Name]?: State[Key] | undefined }, Name, State[Name]> {
```

Not sure about those, though. Just looks better to me.